### ISSUE TITLE
Discuss the diagnostics name for logger

### ISSUE THREAD

**Issue by kenji-miyake** [autowarefoundation/autoware_launch#212]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I've agreed with the maintainers that I can plan this task.

### Description

As described in https://github.com/autowarefoundation/autoware_launch/pull/211#discussion_r1109145181, it's the name of TIER IV's logger.
I concern that it might not be suitable for the AWF because the AWF's configurations/interfaces should be generic and diagnostic names can be said to be interfaces of Autoware.

### Purpose

To make Autoware's interfaces more generic so that people can easily use/extend it.

### Possible approaches

- Rename it to be more generic.
- Remove the diagnostic and add it in the user's (TIER IV's) fork.

### Definition of done

- Reach an agreement among software architects and core developers (@xmfcx @mitsudome-r @yukkysaito @ito-san @asana17 etc.).

**Comment by kenji-miyake** [comment]
@xmfcx @mitsudome-r @yukkysaito @ito-san @asana17 What do you think about it?

**Comment by mitsudome-r** [comment]
Possible candidates:
- rosbag_logger
- data_logger
- autoware_data_logger
- logging

"logger" could be "recorder" as well.

**Comment by ito-san** [comment]
The `bagpacker` and related diagnostics are specific to TIER IV, so I second removing it.

**Comment by asana17** [comment]
I agree with removing the logger from the AWF configuration. It is better to add diagnostics as the general logger function interface is defined.
